src/components/LenderProfile/LenderBadges.vue

```diff
@@ -13,92 +13,48 @@
 			/>
 			<badges-list
 				class="tw-my-4"
-				:completed-achievements="completedAchievements"
+				:completed-achievements="completedBadges"
 				:total-possible-badges="totalPossibleBadges"
 				:is-loading="isLoading"
 			/>
 		</section>
 	</async-lender-section>
 </template>
 
-<script>
-import { gql } from 'graphql-tag';
-import logReadQueryError from '#src/util/logReadQueryError';
+<script setup>
 import BadgesList from '#src/pages/Portfolio/LendingStats/BadgesList';
 import KvLoadingPlaceholder from '#kv-components/KvLoadingPlaceholder';
+import { computed, ref, inject } from 'vue';
+import useBadgeData from '#src/composables/useBadgeData';
 import AsyncLenderSection from './AsyncLenderSection';
 
-const userAchievementProgressQuery = gql`query userAchievementProgress( $publicId: String!) {
-	userAchievementProgress(publicId: $publicId) {
-		id
-		lendingAchievements {
-			id
-			milestoneProgress {
-				id
-				milestoneStatus
-			}
-		}
-	}
-}`;
-
-export default {
-	name: 'LenderBadges',
-	inject: ['apollo', 'cookieStore'],
-	components: {
-		BadgesList,
-		KvLoadingPlaceholder,
-		AsyncLenderSection,
-	},
-	props: {
-		lenderInfo: {
-			type: Object,
-			default: () => ({})
-		},
-		publicId: {
-			type: String,
-			required: true,
-		},
+const props = defineProps({
+	lenderInfo: {
+		type: Object,
+		default: () => ({})
 	},
-	data() {
-		return {
-			isLoading: true,
-			allAchievements: [],
-		};
+	publicId: {
+		type: String,
+		required: true,
 	},
-	computed: {
-		lenderName() {
-			return this.lenderInfo?.name ?? '';
-		},
-		badgesTitle() {
-			return this.lenderInfo?.name
-				? `${this.lenderInfo.name}'s badges`
-				: 'Badges';
-		},
-		completedAchievements() {
-			return this.allAchievements.filter(
-				achievement => achievement.milestoneProgress?.[0]?.milestoneStatus === 'COMPLETE'
-			);
-		},
-		totalPossibleBadges() {
-			return this.allAchievements?.length ?? 0;
-		},
-	},
-	methods: {
-		async fetchUserAchievements() {
-			try {
-				const { data } = await this.apollo.query({
-					query: userAchievementProgressQuery,
-					variables: {
-						publicId: this.publicId,
-					},
-				});
+});
 
-				this.allAchievements = data?.userAchievementProgress?.lendingAchievements ?? [];
-				this.isLoading = false;
-			} catch (e) {
-				logReadQueryError(e, 'LenderBadges userAchievementsProgress');
-			}
-		},
-	},
+const apollo = inject('apollo');
+
+const {
+	badgeData,
+	fetchAchievementData,
+	fetchContentfulData,
+	completedBadges,
+} = useBadgeData(apollo);
+const isLoading = ref(true);
+const badgesTitle = computed(() => (props.lenderInfo?.name ? `${props.lenderInfo.name}'s badges` : 'Badges'));
+
+const fetchUserAchievements = async () => {
+	await fetchAchievementData(apollo, props.publicId);
+	await fetchContentfulData(apollo);
+	isLoading.value = false;
 };
+
+const totalPossibleBadges = computed(() => badgeData.value?.length ?? 0);
```

Comment:
Looking at the screenshot, I don't think this number is correct, it shows `15`. Could you ask product in our team channel if the # of badges should include all of the tiers? If so the number will be much larger.